Add Turkish translations for documentation files

aligunesv opened the pull request:
Translated markdown documentation files into Turkish and organized them under a new 'tr/' directory. This makes it easier for Turkish developers to read, understand, and contribute to the WSL project. No changes were made to the original English documentation.

## Summary of the Pull Request

This PR adds Turkish translations for 57 markdown documentation files. All translations are placed in a new `tr/` folder, with no changes to the original English files.

## PR Checklist

- [ ] **Closes:** Not applicable
- [ ] **Communication:** I have not discussed this with core contributors, but it is a documentation-only addition
- [ ] **Tests:** Not applicable
- [x] **Localization:**  All added content is localized to Turkish
- [ ] **Dev docs:** Not applicable
- [x] **Documentation updated:**  Turkish versions of all major documentation files have been added under `tr/`

## Detailed Description of the Pull Request / Additional comments

This PR introduces a Turkish translation of all key markdown documentation files in the repository. These translations are placed under a `tr/` directory to avoid interfering with the original files. This structure also allows future translations in other languages to follow a similar format (e.g., `fr/`, `de/`, etc.).

Let me know if you'd prefer the translated files to be named differently (e.g., `README.tr.md` instead of placing them under a folder).

## Validation Steps Performed

- Manually verified that all markdown files render properly in GitHub's markdown viewer.
- Confirmed that all links within the translated documents work correctly.

aligunesv committed:
Add Turkish translations for documentation files

Translated markdown documentation files into Turkish and organized them under a new 'tr/' directory.
This makes it easier for Turkish developers to read, understand, and contribute to the WSL project.
No changes were made to the original English documentation.

aligunesv commented:
@microsoft-github-policy-service agree

aligunesv committed:
Merge branch 'master' into master

jsoref commented on tr/CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,9 @@
+# Microsoft Açık Kaynak Davranış Kuralları
+
+Bu proje [Microsoft Açık Kaynak Davranış Kuralları](https://opensource.microsoft.com/codeofconduct/)'nı benimsemiştir .
```
Is the space before `.` intentional? It doesn't seem to fit with the rest of the documents...

aligunesv replied:
No sorry. It's a punctuation mistake.

aligunesv committed:
Update CODE_OF_CONDUCT.md

craigloewen-msft commented:
We need to figure out a story for how we do localization for our developer docs. This PR's intent is spot on, but I think we need to think about how we scale this to other languages as well. I'll ask about this internally!

aligunesv commented:
Thank you for the feedback! I’d be happy to help if there’s a structured localization process in the future. Looking forward to hearing more.

jsoref commented:
Instead of `<lang>/...`, I'd personally put these into one of:
* `doc/<lang>/...` (see https://github.com/microsoft/winget-cli/tree/master/doc)
* `docs/<lang>/...` (see https://github.com/microsoft/pyright/tree/main/docs)
* `translations/<lang>/...` (see https://github.com/microsoft/ML-For-Beginners/blob/09b03ccb52eb73842fd01a3f93a30f2905410bfa/translations/tr/README.md
* `Localization/<lang>/...` (see https://github.com/microsoft/winget-cli/tree/master/Localization)

catuhana commented on tr/doc/docs/technical-documentation/boot-process.md:
```diff
@@ -0,0 +1,117 @@
+
+[wslservice.exe](wslservice.exe.md) çalıştıktan sonra, bu dağıtım içinde linux süreçlerini başlatmak için kullanılabilecek yeni bir [session leader](session-leader.md) başlatmak için bir `LxInitMessageCreateSession` mesajı gönderebilir.
+
+## Linux sürecinin girdi ve çıktılarının Windows'a aktarılması
```
```suggestion
## Linux Sürecinin Girdi ve Çıktılarının Windows'a Aktarılması
```

catuhana commented on tr/doc/docs/technical-documentation/boot-process.md:
```diff
@@ -0,0 +1,117 @@
+[wslservice.exe](wslservice.exe.md) tarafından istenen tüm yapılandırmayı uyguladıktan sonra, sanal makine Linux dağıtımlarını başlatmaya hazırdır.
+
+
+## Bir Linux dağıtımı başlatmak
```
```suggestion
## Bir Linux Dağıtımı Başlatmak
```

catuhana commented on tr/doc/docs/technical-documentation/boot-process.md:
```diff
@@ -0,0 +1,117 @@
+
+Sanal makine başlatıldığında, sağlanan çekirdeğe önyükleme yapacak ve ardından [mini_init](mini_init.md) komutunu çalıştıracaktır.
+
+## Linux önyükleme süreci
```
```suggestion
## Linux Önyükleme Süreci
```

catuhana commented on tr/doc/docs/index.md:
```diff
@@ -0,0 +1,10 @@
+# WSL teknik belgeleri
```
```suggestion
# WSL Teknik Belgeleri
```

catuhana commented on tr/doc/docs/dev-loop.md:
````diff
@@ -0,0 +1,71 @@
+bin\x64\debug\test.bat /name:*UnitTest* -f
+```
+
+## Hata ayıklama testleri
````
```suggestion
## Hata Ayıklama Testleri
```

catuhana commented on tr/doc/docs/dev-loop.md:
```diff
@@ -0,0 +1,71 @@
+
+Bir Hyper-V sanal makinesine dağıtmak için `powershell tools\deploy\deploy-to-vm.ps1 -VmName <vm> -Username <kullanıcı adı> -Password <parola>` kullanabilirsiniz
+
+## Çalışan testler
```
```suggestion
## Çalışan Testler
```

catuhana commented on tr/doc/docs/dev-loop.md:
```diff
@@ -0,0 +1,71 @@
+    - .NET masaüstü geliştirme
+    - .NET WinUI uygulama geliştirme araçları
+    
+## Building WSL
```
Missing translation.

```suggestion
## WSL'yi Oluşturmak
```

catuhana commented on tr/doc/docs/debugging.md:
````diff
@@ -0,0 +1,67 @@
+wsl --debug-shell
+```
+
+Daha sonra `tdnf install gdb` komutunu çalıştırarak `gdb`yi kurabilir ve hata ayıklama işlemlerini başlatabilirsiniz.
````
A small typo in the command.

```suggestion
Daha sonra `dnf install gdb` komutunu çalıştırarak `gdb`yi kurabilir ve hata ayıklama işlemlerini başlatabilirsiniz.
```

catuhana commented on tr/doc/docs/debugging.md:
```diff
@@ -0,0 +1,67 @@
+Bir WSL işleminde gdb ile hata ayıklamanın en basit yolu, koda gdb'den erişmek için `/mnt` bağlama noktalarını kullanmaktır. 
+Başladıktan sonra, kaynak dosyalara bağlanmak için gdb'de `dir /path/to/wsl/source` kullanın.
+
+## Root namespace hata ayıklama
```
```suggestion
## Root Namespace Hata Ayıklama
```

catuhana commented on tr/doc/docs/debugging.md:
```diff
@@ -0,0 +1,67 @@
+
+İşlemler çöktüğünde otomatik olarak çökme dökümlerini toplamak için [bu numarayı](https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#11-reporting-a-wsl-process-crash) da kullanabilirsiniz.
+
+## Linux hata ayıklama
```
```suggestion
## Linux Hata Ayıklama
```

catuhana commented on tr/doc/docs/debugging.md:
```diff
@@ -0,0 +1,67 @@
+dosyasını `%USERPROFILE%/.wslconfig` olarak değiştirin ve WSL'yi yeniden başlatın
+
+
+## Hata ayıklayıcıları ekleme
```
```suggestion
## Hata Ayıklayıcıları Ekleme
```

catuhana commented on tr/doc/docs/technical-documentation/drvfs.md:
Do we want to translate `mount` in this document? Seems to be missing if that's the case.

aligunesv replied:
I didn't want to translate it because it's a technical term. It can be understood differently.

catuhana reviewed (commented):
Looked a bit and found some small things. Will have a better look later. Please correct me if I have wrong reviews.

catuhana commented on tr/CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+
+Kod tamamlandığında günlük dosyasına giden yolun çıktısını verecektir.
+
+#### Otomatik çökme dökümü toplamayı etkinleştirme
```
```suggestion
#### Otomatik Çökme Dökümü Toplamayı Etkinleştirme
```

catuhana commented on tr/CONTRIBUTING.md:
````diff
@@ -0,0 +1,155 @@
+reg.exe delete "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\Windows Error Reporting\LocalDumps" /f
+```
+
+### 12) wslservice için Time Travel Debugging İzleri Toplama
````
```suggestion
### 12) wslservice İçin Time Travel Debugging İzleri Toplama
```

aligunesv committed:
Update tr/CONTRIBUTING.md

Co-authored-by: tuhana <tuhana.cat+github@gmail.com>

aligunesv committed:
Update tr/CONTRIBUTING.md

Co-authored-by: tuhana <tuhana.cat+github@gmail.com>

aligunesv commented:
@catuhana Thank you for your help. I'm going through and organizing each one.

aligunesv commented:
@jsoref Thank you. it would be better to collect them in the `translations` folder. I will organize it this way.

aligunesv committed:
Folder Structure has changed, spelling errors have been fixed.

Fixed some uppercase - lowercase issues.
Fixed some punctuation issues.
Moved files into the 'translations' folder.

aligunesv committed:
Merge branch 'master' of https://github.com/aligunesv/WSL

catuhana commented:
I think it would make more sense to put documentations and its translations under `docs/` folder. `translations/` implies the strings in the code base being translated.

jsoref commented on translations/tr/CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+
+Time Travel Debugging İzleri'ni toplamak için:
+
+1) [Windbg Preview'i yükleyin](https://apps.microsoft.com/store/detail/windbg-preview/9PGJGD53TN86?hl=en-us&gl=us&rtc=1)
```
```suggestion
1) [WinDbg Preview'i yükleyin](https://apps.microsoft.com/store/detail/windbg-preview/9PGJGD53TN86?rtc=1)
```

jsoref commented on translations/tr/CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+
+8) windbg'ye geri dönün ve `Stop and Debug`'a tıklayın
+
+9) İz toplama işlemi tamamlandığında, `Stop Debugging` seçeneğine tıklayın ve Windbg'yi kapatın
```
```suggestion
9) İz toplama işlemi tamamlandığında, `Stop Debugging` seçeneğine tıklayın ve WinDbg'yi kapatın
```

---

I can't speak to Turkish style, you might want to put `` ` `` around `WinDbg`...

jsoref commented on translations/tr/doc/docs/debugging.md:
````diff
@@ -0,0 +1,67 @@
+wpr -stop logs.ETL
+```
+
+Log dosyası kaydedildikten sonra, logları görüntülemek için [WPA](https://apps.microsoft.com/detail/9n58qrw40dfw?hl=en-US&gl=US) adresini kullanabilirsiniz.
````
```suggestion
Log dosyası kaydedildikten sonra, logları görüntülemek için [WPA](https://apps.microsoft.com/detail/9n58qrw40dfw) adresini kullanabilirsiniz.
```

jsoref reviewed (commented):
I'm not a Turkish speaker (and I've only been to Türkiye once)...

My comments are all purely about consistency.

aligunesv committed:
Update translations/tr/CONTRIBUTING.md

Co-authored-by: Josh Soref <2119212+jsoref@users.noreply.github.com>

craigloewen-msft commented:
So there isn't a formalized localization process for the developer docs that are currently on this repo. The challenge here will be scaling to many different languages and maintaining those translations over changes. I don't think an approach of having all the languages in this repo will work well as it will be difficult to maintain (Any dev docs change would need updates across all languages from community members etc.). So I think there are two possible solutions: First you could fork this repo and maintain a translation of the dev docs in Turkish. Or the other possibility would be to use the dev docs at aka.ms/wsldocs which does have localization. The docs on this repo are intended to be lightweight, and tightly coupled to our code which is written and maintained only in English. We [added a new docs page to our public docs](https://github.com/MicrosoftDocs/WSL/blob/main/WSL/opensource.md) which gives an overview of open sourcing, and if there is key information about open sourcing that is needed we can aim to put it there and it will be localized. So I will close this PR as I do not think there is action needed to be take on this specific code base, but happy to hear any feedback and continue the discussion, or review any other PRs on the docs repo etc.. Thank you!
